src/Microsoft.ML.Api/TypedCursor.cs

```diff
@@ -26,6 +26,21 @@ public interface IRow<TRow> : IRow
         void FillValues(TRow row);
     }
 
+    /// <summary>
+    /// This interface is an <see cref="IRow"/> with 'stringly typed' binding.
+    /// It can accept values of type <typeparamref name="TRow"/> and present the value as a row.
+    /// </summary>
+    /// <typeparam name="TRow"></typeparam>
+    public interface IInputRow<TRow> : IRow
```

Comment:
>IInputRow [](start = 21, length = 9)

I want to codify a bit better the difference between `IRow<T>` and `IInputRow<T>`. The difference is mostly in how they are instantiated, right? The first one takes an existing row, and lets us read it into object, and the second one lets us take an object and treat it as a row?

So maybe we could have like `IRowReadableAs<TRow>` and `IRowBackedBy<TRow>` ? or something like that?

Reply:
If you're comfortable changing the API interface signatures that's fine.

---
In reply to: [219580213](https://github.com/dotnet/machinelearning/pull/973#discussion_r219580213) [](ancestors = 219580213)